database/bulkSeedMon.js

```diff
@@ -0,0 +1,51 @@
+const faker = require('faker');
+const generator = require('../helpers/generator.js');
+const db = require('./index.js');
+
+const generateData = () => {
+  return new Promise((resolve) => {
+    const descriptions = [];
+    for (let i = 0; i < 100000; i++) {
+      const noun = generator.genNoun();
+      const numBedrooms = generator.genNumBedrooms(noun);
+      const numGuests = generator.genNumGuests();
+      descriptions.push({
+        host: {
+          _id: i,
+          name: faker.name.firstName(),
+          pic: faker.image.avatar()
+        },
+        title: generator.genTitle(noun),
+        city: generator.genCity(),
+        numGuests,
+        homeType: generator.genHomeType(noun),
+        numBedrooms,
+        numBeds: generator.genNumBeds(numGuests),
+        numBaths: generator.genNumBaths(),
+        description: generator.genDescription(noun, numBedrooms)
+      });
+    }
+    console.log('one: return descriptions');
+    resolve(descriptions);
+  });
+};
+
+const insert = (houses) => {
+  return new Promise((resolve) => {
+    db.Description.insertMany(houses, (error) => {
+      if (error) { throw error; }
+      console.log('two: inserted many houses!');
+      resolve();
+    });
+  });
+};
+
+const seed = async (items) => {
+  for (let i = 0; i < items; i++) {
+    const houses = await generateData();
```

Comment:
A good approach for returning promise to use await syntax on generateData().
One thing which questions me is that generateData() can just simply return a data array so that it won't even need to pause with the await keyword.

Reply:
Yes, I agree with your question. Originally, I had written generateData without using `await`, but then I later included it to be overly cautious and insure that generateData completed before the insert function.